src/transaction/transaction_manager.cpp

```diff
@@ -187,12 +204,9 @@ timestamp_t TransactionManager::Abort(TransactionContext *const txn) {
   LogAbort(txn);
 
   // We hand off txn to GC, however, it won't be GC'd until the LogManager marks it as serialized
-  if (gc_enabled_) {
-    common::SpinLatch::ScopedSpinLatch guard(&timestamp_manager_->curr_running_txns_latch_);
-    // It is not necessary to have to GC process read-only transactions, but it's probably faster to call free off
-    // the critical path there anyway
-    // Also note here that GC will figure out what varlen entries to GC, as opposed to in the abort case.
-    completed_txns_.push_front(txn);
+  // TODO(Ling): eventually we will removed the gc_enabled flag after completely integrate the deferred action framework
+  if (gc_enabled_ && deferred_action_manager_ != DISABLED) {
+    CleanTransaction(txn);
```

Comment:
Doesn't the unlink have to be deferred?

Reply:
Deferring unlinking is in CleanTransaction() function.